controllers/managedocs_controller_test.go

```diff
@@ -1343,7 +1343,13 @@ var _ = Describe("ManagedOCS controller", func() {
 
 				sm := k8sMetricsServiceMonitorTemplate.DeepCopy()
 				smKey := utils.GetResourceKey(sm)
-				Expect(k8sClient.Get(ctx, smKey, sm)).Should(Succeed())
+
+				// Contention of test and reconilcer updating the resource is observed
+				// try to get the resource consistently before updating the resource
+				Consistently(func() error {
+					return k8sClient.Get(ctx, smKey, sm)
```

Comment:
I think you need to make sure that the resource version is stable, without it you don't know if updates are done or not. Right now this get will succeed on the first try and is equivalent to the original line.

Reply:
- Ack, I'm now reading about the matchers.
- Will make the changes as requested

Reply:
- I think it doesn't exist in first successful try
```
In the case of Consistently, Gomega polls the the input repeatedly and asserts the matcher is satisfied every time. Consistently only exits early if a failure occurs - otherwise it continues polling until the specified interval elapses. This is often the only way to assert that something "does not happen" in an asynchronous system.
```